tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/jdt/ui/wizards/NewInterfaceCreationWizardTest.java

```diff
@@ -0,0 +1,178 @@
+
+		String enclosingTypeDefaultValue = interfacePage.getEnclosingType();
+		interfacePage.setEnclosingType(ENCLOSING_TYPE_CUSTOM_VALUE);
+		assertTrue(
+				"Interface enclosing type obtained from the wizard using getEnclosingType() is not the same as the type given using setEnclosingType().",
+				interfacePage.getEnclosingType().equals(ENCLOSING_TYPE_CUSTOM_VALUE));
+		interfacePage.setEnclosingType(enclosingTypeDefaultValue);
+
+		interfacePage.toggleEnclosingTypeCheckBox(false);
+
+		// test adding, verifying and deleting extended interfaces
+		List<String> extendedIterfaceList = new ArrayList<String>();
+		extendedIterfaceList.add("javax.accessibility.AccessibleAction");
+		extendedIterfaceList.add("javax.accessibility.AccessibleEditableText");
+		extendedIterfaceList.add("javax.accessibility.AccessibleExtendedTable");
+		extendedIterfaceList.add("javax.accessibility.AccessibleStreamable");
+		extendedIterfaceList.add("javax.accessibility.AccessibleText");
+
+		for (String extendedInterfaceName : extendedIterfaceList) {
+			interfacePage.addExtendedInterface(extendedInterfaceName);
+		}
+
+		assertTrue(
+				"List of interfaces obtained using function getExtendedInterfaces() has to be the same as list interfaces, that has been set.",
+				interfacePage.getExtendedInterfaces().equals(extendedIterfaceList));
+
+		for (String extendedInterfaceName : extendedIterfaceList) {
+			interfacePage.removeExtendedInterface(extendedInterfaceName);
+		}
+
+		// toggle generate comments checkbox
+		interfacePage.toggleGenerateComments(true);
+		interfacePage.toggleGenerateComments(false);
+
+		// test values in fields
+		assertTrue(
+				"Interface name obtained from the input using getName() function is not the same as name set using setName().",
+				interfacePage.getName().equals(INTERFACE_NAME));
+		assertTrue(
+				"Package name obtained from the input using getPackage() function is not the same as name set using setPackage().",
+				interfacePage.getPackage().equals(PACKAGE));
+		assertTrue(
+				"Source folder name obtained from the input using getSourceFolder() function is not the same as name set using setSourceFolder().",
+				interfacePage.getSourceFolder().equals(SOURCE_FOLDER_NAME));
+
+		// finish the wizard
+		interfaceDialog.finish();
+	}
+
+	@Test
+	public void testAddingAndDeletingNonexistingAndNotFullySpecifiedInterfaces() {
+		// open the wizard
+		NewInterfaceCreationWizard interfaceDialog = new NewInterfaceCreationWizard();
+		interfaceDialog.open();
+		NewInterfaceCreationWizardPage interfacePage = new NewInterfaceCreationWizardPage(interfaceDialog);
+		interfacePage.setName(INTERFACE_NAME + "2");
+		interfacePage.setPackage(PACKAGE);
+		interfacePage.setSourceFolder(SOURCE_FOLDER_NAME);
+
+		// try to add a non-existing extended interface
+		boolean exceptionThrown = false;
+		try {
+			interfacePage.addExtendedInterface("nonexisting.interface.name");
+		} catch (RedDeerException e) {
+			exceptionThrown = true;
```

Comment:
I would rewrite this boolean variable assert to simply call fail if addExtendedInt... passes. Catch block would stay empty.

Reply:
It will be similar in cases below as well.

Reply:
You are right, it would be easier to read. Fixed.